### PR TITLE
test: Integration test admin delete

### PR DESCRIPTION
**Pull request by KIrie-0217**
*Issue #, if available:*
#143 

*Description of changes:*
I have added the test for dy admin delete command.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by KIrie-0217** [comment]
During the execution of the destructor for tm, cleanup is performed on a table that has already been deleted, which causes an error. 
To avoid this, the destructor's processing is skipped using std::mem::forget. 
While this is not unsafe, the use of std::mem::forget may potentially lead to memory leaks in the future.

I would like you to evaluate whether this implementation is valid or not.

https://github.com/KIrie-0217/dynein/blob/integration-test-admin-delete/tests/admin_delete.rs#L64-L65

File: tests/util/mod.rs [hunk 1]
```diff
@@ -137,6 +137,20 @@ impl<'a> TestManager<'a> {
...
         Ok(())
     }
+
+    /// Remove a temporary table from the list of temporary tables.
+    pub fn remove_temporary_table(
+        &mut self,
+        table_name: String,
+    ) -> Result<(), Box<dyn std::error::Error>> {
+        if self.temporary_tables.contains(&table_name) {
+            self.temporary_tables.retain(|s| !s.contains(&table_name));
```
Comment (StoneDot) [inline tests/util/mod.rs:147]: I think the following code is more suitable for this case.

```suggestion
            self.temporary_tables.retain(|s| s != &table_name);
```

File: tests/util/mod.rs [hunk 2]
```diff
@@ -137,6 +137,20 @@ impl<'a> TestManager<'a> {
 
         Ok(())
     }
+
+    /// Remove a temporary table from the list of temporary tables.
+    pub fn remove_temporary_table(
+        &mut self,
+        table_name: String,
+    ) -> Result<(), Box<dyn std::error::Error>> {
+        if self.temporary_tables.contains(&table_name) {
```
Comment (StoneDot) [inline tests/util/mod.rs:146]: I think this check is unnecessary. If you want to ensure the deletion, we should check the change of the length of `self.temporary_tables`.

File: tests/util/mod.rs [hunk 3]
```diff
@@ -137,6 +137,20 @@ impl<'a> TestManager<'a> {
 
         Ok(())
     }
+
+    /// Remove a temporary table from the list of temporary tables.
+    pub fn remove_temporary_table(
+        &mut self,
+        table_name: String,
+    ) -> Result<(), Box<dyn std::error::Error>> {
```
Comment (StoneDot) [inline tests/util/mod.rs:145]: This function always succeeds. We do not need to use `Result`.

**Review by StoneDot — COMMENTED** [review]
Thank you for adding tests for `dy admin delete` command. I left some comments. Could you please check them?

File: tests/util/mod.rs (hunk 2 above)
Comment (KIrie-0217) [reply tests/util/mod.rs:146]: I felt that determining whether a temporary table has been deleted or not should not be done in this function, so I will remove that processing

File: tests/util/mod.rs (hunk 1 above)
Comment (KIrie-0217) [reply tests/util/mod.rs:147]: I agree

File: tests/util/mod.rs (hunk 3 above)
Comment (KIrie-0217) [reply tests/util/mod.rs:145]: I think the return value is unnecessary, so I will remove it.

File: tests/util/mod.rs [hunk 4]
```diff
@@ -199,6 +199,11 @@ impl<'a> TestManager<'a> {
 
         Ok(())
     }
+
+    /// Remove a temporary table from the list of temporary tables.
+    pub fn remove_temporary_table(&mut self, table_name: String) {
+        self.temporary_tables.retain(|s| s != &table_name);
```
Comment (StoneDot) [inline tests/util/mod.rs:205]: nit: You can improve the interface using `AsRef<str>`. But, it is not necessary. 

```suggestion
    pub fn remove_temporary_table<S: AsRef<str>>(&mut self, table_name: S) {
        let table_name = table_name.as_ref();
        self.temporary_tables.retain(|s| s != table_name);
```

File: tests/admin_delete.rs [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+        "--yes",
+    ]);
+    cmd.assert().success().stdout(format!(
+        "Delete operation for the table '{}' has been started.\n",
+        table_name
+    ));
+
+    // To prevent double deletion in the Drop trait, exclude the table here
+    tm.remove_temporary_table(table_name);
+
```
Comment (StoneDot) [inline tests/admin_delete.rs:65]: bonus: Is there a good way to check whether the table was deleted after the command? If so, it would be excellent.

**Review by StoneDot — APPROVED** [review]
Thank you for polishing up the code. Now, LGTM. I left some comments for additional improvement. If you'd like to include them in this pull request, I'll check them after you change them.

File: tests/admin_delete.rs (hunk 5 above)
Comment (KIrie-0217) [reply tests/admin_delete.rs:71]: I agrre, I'll add the logic

File: tests/util/mod.rs [hunk 6]
```diff
@@ -199,6 +199,15 @@ impl<'a> TestManager<'a> {
...
     }
+
+    /// Remove a temporary table from the list of temporary tables.
+    pub fn remove_temporary_table(&mut self, table_name: &String) {
+        self.temporary_tables.retain(|s| s != table_name);
+    }
+
+    pub fn count_temporary_tables(&mut self) -> usize {
+        self.temporary_tables.len()
+    }
```
Comment (StoneDot) [inline tests/util/mod.rs:210]: This function is not used.

**Review by StoneDot — COMMENTED** [review]
I think added logic is reasonable. Could you remove the unused function `count_temporary_tables`?

File: tests/util/mod.rs (hunk 6 above)
Comment (KIrie-0217) [reply tests/util/mod.rs]: I'll delete it

**Review by StoneDot — APPROVED** [review]
Thank you for improving the pull request. All looks great to me.

**Review by StoneDot — APPROVED** [review]

**Pull request merged**